lib/periodicChecks.js

```diff
@@ -0,0 +1,234 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Handler for checks to be carried out periodically.
+ */
+
+const {
+  getAllOpenPullRequests,
+  hasPullRequestBeenApproved,
+  getChangelogLabelFromPullRequest,
+  getProjectOwnerFromLabel,
+  doesPullRequestHaveChangesRequested,
+  isUserAMemberOfTheOrganisation,
+} = require('./utils');
+const mergeConflictModule = require('./checkMergeConflicts');
+const { teamLeads } = require('../userWhitelist.json');
+
+/**
+ *  This function checks if a pull request has merge conflicts.
+ *
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const hasMergeConflict = (pullRequest) => {
+  return pullRequest.mergeable === false;
+};
+
+/**
+ *  This function checks if a pull request has pending reviews.
+ *
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const hasPendingReviews = (pullRequest) => {
+  return pullRequest.requested_reviewers.length > 0;
+};
+
+/**
+ * This function pings and assigns all pending reviewers to a pull request.
+ * @param {import('probot').Context} context
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const pingAndAssignReviewers = async (context, pullRequest) => {
+  const reviewers = pullRequest.requested_reviewers.map(
+    (reviewer) => reviewer.login
+  );
+
+  context.github.issues.addAssignees(
+    context.repo({
+      issue_number: pullRequest.number,
+      assignees: reviewers,
+    })
+  );
+
+  context.github.issues.createComment(
+    context.repo({
+      issue_number: pullRequest.number,
+      body:
+        'Assigning @' +
+        reviewers.join(', @') +
+        ' for codeowner reviews. Thanks!',
+    })
+  );
+};
+
+/**
+ * This function assigns the PR author to a pull request.
+ * @param {import('probot').Context} context
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ * @param {string} comment
+ */
+const pingAndAssignPrAuthor = async (context, pullRequest, comment) => {
+  context.github.issues.addAssignees(
+    context.repo({
+      issue_number: pullRequest.number,
+      assignees: [pullRequest.user.login],
+    })
+  );
+  await context.github.issues.createComment(
+    context.repo({
+      issue_number: pullRequest.number,
+      body: comment,
+    })
+  );
+};
```

Comment:
My one concern with all this is aren't we doing similar tasks in #116, shouldn't we reuse the common functions?

Reply:
Yes, we are doing something similar there.
We can't reuse the function since #116 has not been merged.

Reply:
This function is still valuable after merging #116.